code/modules/mob/living/carbon/superheroes.dm

```diff
@@ -14,15 +14,23 @@
 	assign_genes(H)
 	assign_spells(H)
 	equip(H)
+	fixflags(H)
 	assign_id(H)
 
 /datum/superheroes/proc/equip(var/mob/living/carbon/human/H)
 	H.rename_character(H.real_name, name)
 	for(var/obj/item/W in H)
 		if(istype(W,/obj/item/organ)) continue
+		if(istype(W,/obj/item/weapon/implant)) continue
 		H.unEquip(W)
 	H.equip_to_slot_or_del(new /obj/item/device/radio/headset(H), slot_l_ear)
 
+/datum/superheroes/proc/fixflags(var/mob/living/carbon/human/H)
+	for(var/obj/item/W in H)
```

Comment:
I think you can avoid the `if(istype` checks by making this loop through `H.get_all_slots()` instead of plain `H`